getting_started/scripting/gdscript/gdscript_documentation_comments.rst

```diff
@@ -0,0 +1,219 @@
+    ## 
+    ## @tutorial:            http://the/tutorial1/url.com
+    ## @tutorial(Tutorial2): http://the/tutorial2/url.com
+    ##
+
+.. warning:: If there is any space in between the tag name and colon ``@desc  :``
+             it won't treated as a valid tag and ignored.
+
+
+.. note:: When the description is multi line the preceding and trailing white
+          spaces will be striped and joined with a single space. If the line 
+          break (or any other alignment) is intended use
+          :ref:`bbcode <doc_bbcode_in_richtextlabel>`.
+
+Documenting script members
+--------------------------
+
+Documentation of a script member must immediately precede the member or it's
+annotations if it has any **except for enum values** which's description is should
+be at the the same line as it is. It's designed like this for a better readability
+and look. The description could have more than one line but every line must starts
+with the double hash symbol ``##`` to be considered as it's part of the documentation.
+The script documentation will update in the editor help window every time the
+script updated and if any member variable or function name starts with an
+underscore and undocumented it'll treated as private for the documentation and
+ignored in the in the help window.
+
+Members that are applicable for the documentation
+
+- Inner class
+- Constant
+- Function
+- Signal
+- Variable
+- Enum
+- Enum value
+
+Examples
+--------
+
+::
+
+    extends Node2D
+	
+    ## 
+    ## A brief description about your script.
+    ##
+    ## @desc:
+    ##     The description about the script, what it
+    ##     can do, how to do that and everything.
+    ## 
+    ## @tutorial:            http://the/tutorial1/url.com
+    ## @tutorial(Tutorial2): http://the/tutorial2/url.com
+    ##
+    
+    ## the description about the variable v1
+    var v1
+    
+    ## multi line description about the variable v2. the type
+    ## information will be extracted below for the documentation
+    var v2: int
+    
+    ## if there are any annotation the member has, then it should
+    ## immediately precede them.
+    @export
+    @onready
+    var v3 := some_func()
+    
+    ## the description of a constant
+    const GRAVITY = 9.8
+	
+    ## the description of a signal
+    signal my_signal
+    
+    ## it's the enum description and see below the exception for 
+    ## the enum values which are documented on the same line.
+    enum Direction {
+        UP    = 0,  ## direction up
+        DOWN  = 1,  ## direction down
+        LEFT  = 2,  ## direction left
+        RIGHT = 3,  ## direction right
+    }
+    
+    ## since the below function is documented even it's name starts with
+    ## an underscore, it'll be updated in the editor.
+    func _fn(p1:int, p2:String) -> int:
+        return 0
+    
+    # the below function isn't documented and starts with an underscore name
+    # so this will treated as private and will not be updated in the editor.
```

Comment:
```suggestion
    # The below function isn't documented and its name starts with an underscore 
    # so it will treated as private and will not be shown in the help window.
```